extensions/wikia/Search/classes/Services/AbstractSearchService.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Wikia\Search\Services;
+
+
+abstract class AbstractSearchService {
+
+	public function query( $phrase ) {
+		$select = $this->prepareQuery( $phrase );
+
+		$response = $this->select( $select );
+
+		return $this->consumeResponse( $response );
+	}
+
+	protected function prepareQuery( $query ) {
+		// Subclass implements
+
+		return null;
```

Comment:
Possibly throw an exception if this `prepareQuery()` is called. Same with all the "abstract" methods in this class.